bin/write_viral_gff.py

```diff
@@ -21,136 +20,213 @@ def get_ena_contig_mapping(ena_contig_file):
-                )
-                #   start with: ERZ2271866.1-NODE-1-length-21396-cov-5.122534	ViPhOG	proviral_region	1020	2050	.	-	.
-                gff.write(
-                    f'{contig_name}\t{viphog}\tviral_sequence\t{row["Start"]}\t{row["End"]}\t.\t{direction}'
-                    f"\t.\t{annotation}\n"
-                )
-    return contigs
-
-
-def write_metadata(
-    checkv_files, taxonomy_files, sample_prefix, virify_contigs, ena_mapping=None
+                    contig_protID[Contig].append(CDS_ID)
+
+                Direction=Direction.replace('-1','-').replace('1','+')
+
+                if not Best_hit == 'No hit':
+                    Best_hit=Best_hit.replace('.faa','')
+                    Label=line_l[7]
+                    annotation='viphog='+Best_hit+';'+'viphog_taxonomy='+Label
+                #else:
+                #    annotation='product=hypothetical_protein'
+
+                    values=(str(Start),str(End),Direction,annotation)
+                    prot_data[CDS_ID]=values
+
+    #return dictionaries
+    return contig_protID, prot_data, contig_data
+
+
+def write_gff(
+    checkv_files, taxonomy_files, sample_prefix, assembly_file, annot_dicts, ena_mapping=None
 ):
+
     if ena_mapping:
         ena_assembly_accession = list(ena_mapping.values())[0].split(".")[0]
-        output_filename = f"{ena_assembly_accession}_virify_contig_viewer_metadata.tsv"
+        output_filename = f"{ena_assembly_accession}_virify.gff"
     else:
-        output_filename = f"{sample_prefix}_virify_contig_viewer_metadata.tsv"
-    headers = (
-        "sequence_id\tcontig\tvirify_taxonomy\tstart_of_first_viphog\tend_of_last_viphog\tcheckv_provirus\t"
-        "checkv_quality\tmiuvig_quality\n"
-    )
-    checkv_dict, taxonomy_dict = {}, {}
+        output_filename = f"{sample_prefix}_virify.gff"
 
+    checkv_dict, taxonomy_dict, contigs_len_dict = {}, {}, {}
     #   parse checkv for quality
     for cfile in checkv_files:
-        checkv_df = pd.read_csv(cfile, sep="\t")
-        for index, row in checkv_df.iterrows():
-            if row["contig_id"] in virify_contigs:
-                checkv_type = row["provirus"]
-                checkv_dict[row["contig_id"]] = {
-                    "checkv_type": checkv_type,
-                    "checkv_quality": row["checkv_quality"],
-                    "miuvig_quality": row["miuvig_quality"],
-                }
+        with open(cfile,'r') as input_table:
+            next(input_table)
+            for line in input_table:
+                line_l=line.rstrip().split('\t')
+                contig_id=line_l[0]
+                checkv_type=line_l[2]
+                checkv_quality=line_l[7]
+                miuvig_quality=line_l[8]
+                value='checkv_provirus='+checkv_type+';checkv_quality='+checkv_quality+';miuvig_quality='+miuvig_quality
+                checkv_dict[contig_id]=value
 
     #   parse taxonomic lineage when available
```

Comment:
Can you improve the comment please, what is the output of this method as it's long and hard to understand.

An input and output in the comment would be enough to undernstand:
```
# Convert the taxonomy from XX to YY
```

Reply:
done!